shell/crates/shell_state/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+        Self {
+            ..Default::default()
+        }
+    }
+}
+
+pub struct ShellStateManager {}
+
+impl ShellStateManager {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn run(cx: &mut App) {
+        let (mut message_tx, mut message_rx) = mpsc::channel::<ShellStateMessage>(10);
+        cx.spawn(async move |app| {
+            while let Some(message) = message_rx.next().await {
+                _ = app.update(|cx| {
+                    match message {
+                        ShellStateMessage::WirelessStatusChanged { enabled } => {
+                            ShellState::global_mut(cx).wireless_details.enabled = enabled;
+                            if !enabled {
+                                ShellState::global_mut(cx)
+                                    .wireless_details
+                                    .connected_network = None;
+                            }
+                        }
+                        ShellStateMessage::WirelessStrength { strength } => {
+                            if let Some(mut connected_network) = ShellState::global(cx)
+                                .wireless_details
+                                .connected_network
+                                .clone()
+                            {
+                                connected_network.signal_strength = strength;
+                                ShellState::global_mut(cx)
+                                    .wireless_details
+                                    .connected_network = Some(connected_network);
+                            }
+                        }
+                        ShellStateMessage::BluetoothEnabled { enabled } => {
+                            ShellState::global_mut(cx).bluetooth_details.enabled = enabled;
+                        }
+                        ShellStateMessage::BluetoothDevices { count } => {
+                            ShellState::global_mut(cx)
+                                .bluetooth_details
+                                .connected_devices = count;
+                        }
+                        ShellStateMessage::BatteryStateChanged { state } => {
+                            ShellState::global_mut(cx).battery_state = state;
+                        }
+                        ShellStateMessage::BatteryLevelChanged { level } => {
+                            ShellState::global_mut(cx).battery_level = level;
+                        }
+                        ShellStateMessage::BatteryPercentageChanged { value } => {
+                            ShellState::global_mut(cx).battery_percent = value;
+                        }
+                        ShellStateMessage::TimeUpdated => {
+                            let current_time_date = get_current_datetime();
+                            ShellState::global_mut(cx).current_time_date = current_time_date;
+                            if let Some(entity) = ShellState::global(cx).status_bar_entity {
```

Comment:
Lets discuss this approach, i am not sure if we should do this?

Also we can think of better name for this crate, shell_state seems a bit weird.